buildpacks/nodejs-yarn/src/layers/dist.rs

```diff
@@ -0,0 +1,96 @@
+use crate::{NodeJsYarnBuildpack, NodeJsYarnBuildpackError};
+use heroku_nodejs_utils::inv::Release;
+use libcnb::build::BuildContext;
+use libcnb::data::buildpack::StackId;
+use libcnb::data::layer_content_metadata::LayerTypes;
+use libcnb::layer::{ExistingLayerStrategy, Layer, LayerData, LayerResult, LayerResultBuilder};
+use libcnb::Buildpack;
+use libherokubuildpack::{decompress_tarball, download_file, log_info, move_directory_contents};
+use serde::{Deserialize, Serialize};
+use std::path::Path;
+use tempfile::NamedTempFile;
+use thiserror::Error;
+
+/// A layer that downloads and installs the yarn cli
+pub struct DistLayer {
+    pub release: Release,
+}
+
+#[derive(Deserialize, Serialize, Clone, PartialEq, Eq)]
+pub struct DistLayerMetadata {
+    layer_version: String,
+    yarn_version: String,
+    stack_id: StackId,
+}
+
+#[derive(Error, Debug)]
+pub enum DistLayerError {
+    #[error("Couldn't create tempfile for Node.js distribution: {0}")]
```

Comment:
should this and the following errors reference `Yarn` instead of `Node.js`?

Reply:
Hah, yeah. I'm still iterating pretty heavily on this.. lots of WIP =P